code/object/objectsnd.cpp

```diff
@@ -577,7 +578,8 @@ void obj_snd_do_frame()
 
 		go_ahead_flag = TRUE;
 		float max_vol,new_vol;
-		if (!osp->instance.isValid()) {
+		// start playing if sound is invalid and looping or non-looping but has not started
+		if (!osp->instance.isValid() && (!(osp->flags & OS_LOOPING_DISABLED) || !(osp->flags & OS_STARTED_PLAYING))) {
```

Comment:
Upon further consideration, this if() block should be augmented a bit.  We want non-looping sounds that have already played once to be removed from the object sound list.  So I suggest something like this:
```
if (!osp->instance.isValid() && (osp->flags & OS_LOOPING_DISABLED) && (osp->flags & OS_STARTED_PLAYING)) {
    list_remove(&obj_snd_list, osp);
else if  (!osp->instance.isValid()) {
```
Not 100% sure that's correct, but I think it's close.

Reply:
This was not only insufficient, it was in the wrong place.  Ended up rewriting this a bit.